akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolFlow.scala

```diff
@@ -77,14 +77,20 @@ private object PoolFlow {
       import GraphDSL.Implicits._
 
       val conductor = b.add(PoolConductor(maxConnections, pipeliningLimit, log))
-      val slots = Vector
-        .tabulate(maxConnections)(PoolSlot(_, connectionFlow, settings))
-        .map(b.add)
+      //The difference between hot and cold slots:
+      // - hot: should be kept alive, the number is based on "minConnections" from settings
+      // - cold: hold ordinary connection that might be killed, the number is "maxConnections" - "minConnections"
+      val hotSlots = Vector
+        .tabulate(minConnections)(PoolSlot(_, connectionFlow, settings, isHotConnection = true))
+      val coldSlots = Vector
+        .tabulate(maxConnections - minConnections)(PoolSlot(_, connectionFlow, settings, isHotConnection = false))
+      val allSlots = (hotSlots ++ coldSlots).map(b.add)
```

Comment:
Good question @ktoso Here's a link on a (bigger) discussion https://issues.scala-lang.org/browse/SI-7725
I think we could make it faster by simply changing the order - 'hotSlots' will (usually) be either smaller or empty, which would allow JVM to short-circuit during the addition. 
Would this be enough to answer the concern?